### PR TITLE
Refactor server side code #25

### LINKED ISSUE #25: Creating src folder and refactoring server-side code
1. Create `src` folder and putting server-side coding files into there.
2. Refactoring server-side code:
    1. According to Airbnb base rules
    2. More modular folder structure and and creating config files.

### PR DESCRIPTION
**Pull request by TheCandiko**
Major refactor! :neckbeard:

1. Refactored server.js files according to AirBnB base rules.
2. Modified project folder structure. For changes see commit [bf71834](https://github.com/chingu-voyage4/Geckos-Team-20/commit/bf7183457333c0ba3f13ccc5af57fc20999be899)
3. Installation instructions added to README.md

I tried to write descriptive commits as I can. Check them out!
If any point is not clear, please let me know, I'll explain.

Closes #25

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -1,28 +1,32 @@
...
-  express.static(__dirname + "/node_modules/bootstrap/dist/")
+  express.static(`${__dirname}/node_modules/bootstrap/dist/`)
 );
 
 // Root Route
-app.get("/", function(req, res) {
-  //res.sendFile(path.join(__dirname,'public', 'index.html'));
-  res.sendFile(path + "/public/index.html");
-});
+app.get("/", (req, res) => res.sendFile(`${path}/public/index.html`));
```
Comment (rajpranz) [inline server.js]: The usage of arrow functions does make the syntax look more concise and easier to read

File: server.js [hunk 2]
```diff
@@ -1,28 +1,32 @@
...
+app.get("*", (req, res) => res.sendFile(`${path}/app/public/notfound.html`));
 
-var port = process.env.PORT || 8080;
-app.listen(port, function() {
-  console.log("Node.js listening on port " + port + "...");
-});
+app.listen(port, () => console.log(`Node.js listening on port ${port}...`));
+
+/*
+no-var rules: all var keywords changed to const keyword
```
Comment (rajpranz) [inline server.js]: Thanks for adding these links. Will have a look at them. They are quite a handy reference.
As you had seen, although I had setup the linter, I had not been actually using it, as you would have noticed. 
Thanks for refactoring the code

File: package.json [hunk 3]
```diff
@@ -23,6 +21,7 @@
   "homepage": "https://github.com/chingu-voyage4/Geckos-Team-20#readme",
   "dependencies": {
     "bootstrap": "^4.0.0",
+    "dotenv": "^5.0.1",
     "express": "^4.16.3",
     "jquery": "^3.3.1"
   },
```
Comment (rajpranz) [inline package.json:27]: We are not using any templating engine at the moment. We might like to implement one at some stage

**Review by rajpranz — APPROVED** [review]
Good you refactored the back end to make things more modular. 
Also you did the refactor to match airbnb rules. I was not running the linter on my machine so had not done so.
Next time you can ask me to make changes in my pull request if there is any issues.

File: package.json (hunk 3 above)
Comment (TheCandiko) [reply package.json:27]: If we need one, which one you are familiar with?

**Comment by TheCandiko** [comment]
@dragonraij 

Thanks for review! If is there any point that is needed to explain, let me know.

I didn't notice the AirBnB rules issue until I've approved and run in my local dev env. Next time I'll be inform before pull request!

**Pull request merged**